src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +320,1319 @@ const ActorInit En_Egol_InitVars = {
+            Math_SmoothStepToS(&this->headRot, -0x2710, 5, 0x1F4, 5);
+        } else {
+            Math_SmoothStepToS(&this->headRot, 0, 5, 0x1F4, 5);
+        }
+        if (this->laserState == EYEGORE_LASER_OFF) {
+            if (this->laserCount >= 3) {
+                EnEgol_SetupWalk(this);
+            } else {
+                this->laserRot.x = sLaserAngles[this->laserCount];
+                this->laserScaleTarget.z = 0.0f;
+                this->laserScale.z = 0.0f;
+                if (this->laserCount == 0) {
+                    this->laserState = EYEGORE_LASER_START;
+                } else {
+                    this->laserState = EYEGORE_LASER_FIRE;
+                    this->laserScaleTarget.x = 0.03f;
+                    this->laserScaleTarget.y = 0.03f;
+                }
+                this->laserCount++;
+            }
+        } else if (this->laserState >= EYEGORE_LASER_FIRE) {
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_EYEGOLE_BEAM - SFX_FLAG);
+            if (this->laserState != EYEGORE_LASER_OFF) {
+                EnEgol_DestroyBlocks(this, play, this->laserCollider.dim.quad[0], this->laserCollider.dim.quad[1]);
+            }
+            if ((this->actionTimer == 0) && BgCheck_EntityLineTest1(&play->colCtx, &this->laserCollider.dim.quad[3],
+                                                                    &this->laserCollider.dim.quad[1], &hitPos, &colPoly,
+                                                                    false, true, false, true, &bgId)) {
+                //! @bug this should check walls, too
+                Vec3s rotToNorm;
+                f32 nx;
+                f32 ny;
+                f32 nz;
+                s32 pad3;
+                s32 i;
+                s32 quakeSize;
+                Player* player = GET_PLAYER(play);
+
+                nx = COLPOLY_GET_NORMAL(colPoly->normal.x);
+                ny = COLPOLY_GET_NORMAL(colPoly->normal.y);
+                nz = COLPOLY_GET_NORMAL(colPoly->normal.z);
+
+                rotToNorm.x = -func_80086B30(-nz * ny, 1.0f) * 0x8000 / M_PI;
+                rotToNorm.z = func_80086B30(-nx * ny, 1.0f) * 0x8000 / M_PI;
+                //! @bug This is the wrong calculation for the angles. It should be
+                // rotToNorm.x = func_80086B30(nz, ny) * 0x8000 / M_PI;
+                // rotToNorm.z = func_80086B30(-nx, sqrtf(1.0f - SQ(nx))) * 0x8000 / M_PI;
+                if ((this->actor.world.pos.y - 50.0f) <= player->actor.world.pos.y) {
+                    EnEgol_SpawnEffect(this, &hitPos, &rotToNorm, 100, 0.02f, EYEGORE_EFFECT_IMPACT);
+                }
+                quakeSize = 4 - (s32)(fabsf(player->actor.world.pos.y - this->actor.world.pos.y) * 0.02f);
+                if (quakeSize > 4) {
+                    quakeSize = 4;
+                } else if (quakeSize < 1) {
+                    quakeSize = 1;
+                }
```

Comment:
Would the `CLAMP` macro work here? There are three instances in this file of this same clamp as well

Reply:
Nope. However, I'm starting to suspect this was originally written as a one-liner: `if (quakeSize > 4) quakeSize = 4; else if (quakeSize < 1) quakeSize = 1;`, which does match. There may even have been a macro for it given how often these self-assigned clamps don't match with the actual CLAMP macros.